app/src/main/java/org/p2p/wallet/send/ui/search/adapter/SearchAdapter.kt

```diff
@@ -13,12 +13,29 @@ class SearchAdapter(
 
     private val data = mutableListOf<SearchResult>()
 
-    /* TODO: Will add diff later */
-    @SuppressLint("NotifyDataSetChanged")
+    private class SearchAdapterDiffUtil(
+        private val oldList: List<SearchResult>,
+        private val newList: List<SearchResult>
+    ) : DiffUtil.Callback() {
+
+        override fun getOldListSize(): Int = oldList.size
+
+        override fun getNewListSize(): Int = newList.size
+
+        override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+            return oldList[oldItemPosition] == newList[newItemPosition]
+        }
+
+        override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
+            return oldList[oldItemPosition] === newList[newItemPosition]
```

Comment:
this should compare addresses only, since they are unique.
Otherwise this will return false and the item will be replaced by a new one